src/components/Checkbox/Checkbox.tsx

```diff
@@ -38,37 +44,57 @@ export class Checkbox extends React.Component<ICheckboxProps, ICheckboxState> {
     }
   }
 
+  public componentWillUnmount() {
+    this._checkboxInput.removeEventListener('focus', this._onFocus.bind(this));
+    this._checkboxInput.removeEventListener('blur', this._onBlur.bind(this));
+  }
+
   public render() {
     let { text, isEnabled, className } = this.props;
     let { isChecked } = this.state;
 
     return (
-      <div className={ css('ms-ChoiceField', className) }>
+      <div
+        className={ css('ms-Checkbox', className) }
+        ref={ (c): HTMLElement => this._checkbox = c }
+      >
         <input
-          ref={ (c): HTMLInputElement => this._checkBox = c }
+          ref={ (c): HTMLInputElement => this._checkboxInput = c }
           id={ this._id }
           name={ this._id }
-          className='ms-ChoiceField-input'
+          className='ms-Checkbox-input'
           type='checkbox'
           role='checkbox'
           checked={ isChecked }
           disabled={ !isEnabled }
           onChange={ this._handleInputChange }
           aria-checked={ isChecked }
```

Comment:
@cschlechty I assume it would be the same in case of a true radio, so I could remove it from ChoiceGroup too?